src/rules/unbalanced-delimiters.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * @license
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+import {Rule} from '../rule';
+import {Document} from 'polymer-analyzer/lib/model/model';
+import {ParsedHtmlDocument} from 'polymer-analyzer/lib/html/html-document';
+import {Severity, Warning} from 'polymer-analyzer/lib/warning/warning';
+import {PolymerElement} from 'polymer-analyzer/lib/polymer/polymer-element';
+import * as dom5 from 'dom5';
+import * as parse5 from 'parse5';
+import * as matchers from '../matchers';
+import * as utils from '../utils';
+
+export class UnbalancedDelimiters implements Rule {
+  public async check(document: Document): Promise<Warning[]> {
+    let warnings: Warning[] = [];
+
+    const parsedHtml = document.parsedDocument;
+
+    // This linter is only relevant to html documents, so exit out unless the
+    // provided document is one.
+    if (!(parsedHtml instanceof ParsedHtmlDocument)) {
+      return warnings;
+    }
+
+    // Address binding expressions only for polymer elements.
+    for (const element of document.getByKind('polymer-element')) {
```

Comment:
Would it be possible to structure this rule with a `getDataBindingExpressions` method? That'd be really broadly useful.

Hm, I guess not, as we're looking for stuff that's only *almost* legal data binding expressions.